state.c

```diff
@@ -444,7 +444,7 @@ int pv_state_validate(struct pv_state *s)
 
 static int pv_state_mount_bsp_volumes(struct pv_state *s)
 {
-	pv_disk_mount_swap(s);
+	pv_disk_mount_swap(&s->disks);
```

Comment:
still think that we should return -1 here in case swap is specified but the mount failed for some reason

Reply:
OK, but currently the function `pv_disk_mount_swap()` only fails if the disks list is empty and only show a log message when a swap disk cannot be mounted.
```C
int pv_disk_mount_swap(struct dl_list *disks)
{
	if (!disks)
		return -1;

	pv_log(INFO, "mounting all swap disk");

	struct pv_disk *d, *tmp;
	dl_list_for_each_safe(d, tmp, disks, struct pv_disk, list)
	{
		if (d->type != DISK_SWAP)
			continue;

		int err = pv_disk_mount(d);
		if (err != 0)
			pv_log(ERROR, "cannot mount %s", d->name);
	}

	return 0;
}
```
So, if I understand right and trying to give sense to this, the function should return -1 if any disk fails?

Reply:
if something is in the specification and it fails for whatever reason, we are not running what the specification contains. So yes, we should return -1 if something fails in pv_disk_mount_swap and then return -1 in pv_state_mount_bsp_volumes too. That way we'll get a rollback in case a disk mount fails.

Reply:
fixed :)